spl/src/metadata.rs

```diff
@@ -183,3 +184,34 @@ pub struct MintNewEditionFromMasterEditionViaToken<'info> {
     //
     pub metadata_mint: AccountInfo<'info>,
 }
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MetadataAccount(mpl_token_metadata::state::Metadata);
+
+impl MetadataAccount {
+    pub const LEN: usize = mpl_token_metadata::state::MAX_METADATA_LEN;
+}
+
+impl anchor_lang::AccountDeserialize for MetadataAccount {
+    fn try_deserialize_unchecked(buf: &mut &[u8]) -> anchor_lang::Result<Self> {
+        mpl_token_metadata::state::Metadata::deserialize(buf)
```

Comment:
Probably want to close this in favor of https://github.com/coral-xyz/anchor/pull/2014/files which uses the `safe_deserialize` variant.